### PR TITLE
bring multichannel support to run_block

### PR DESCRIPTION
**Pull request by philippwindischhofer**
Works already:
* different input voltage ranges for each channel
* better selection of timebase (either manual, or the fastest one consistent with the currently enabled channels)

To be included:
* better interface for function generator that also allows to control the triggering

PS: I forgot to edit the git config again ... so it's you who authored all the commits :)

### CODE REVIEW

**Comment by philippwindischhofer** [comment]
Hi @fgrosa this is now tested & ready!

File: utils.py [hunk 1]
```diff
@@ -119,40 +129,138 @@ def generate_signal(status, chandle, func='PICO_SINE', **kwargs):
...
-        chandle,
+        handle,
         0 if dummy else 1,
         source,
-        mV2adc(trigger_thrs_mV, channel_range, max_ADC),
+        mV2adc(trigger_thrs_mV, pico_channel_range, max_ADC),
         pico_direction,
         0,  # delay = 0 s
-        1000000  # autoTriggerMicroSeconds
+        0  # autoTriggerMicroSeconds
```
Comment (fgrosa) [inline utils.py]: is it ok to put 0 microseconds here? Is it needed to reduce the dead time?

Comment (philippwindischhofer) [reply utils.py]: Aha! When I tested the advanced triggering functionality, I found that setting an autotrigger timeout of zero makes it wait indefinitely (this is consistent with the documentation). I then tried whether `setSimpleTrigger` results in the same behaviour, and it does. (Nothing is stated in the documentation, except that `setSimpleTrigger` calls internally the functions that set up the "advanced" trigger---so that seems consistent.)

Long story short, I turned it into a configurable parameter now, which should be better in any case.

Comment (fgrosa) [reply utils.py]: Thanks for the explanation! 👍

**Comment by fgrosa** [comment]
> Hi @fgrosa this is now tested & ready!

Hi @philippwindischhofer thanks a lot for the huge work! I went through it today and it looks fine for me, but before merging if you want you could try to change the author of the commits with

```
 git -c user.name="New Author Name" -c user.email=email@address.com commit --amend --reset-author
```

so that we keep track of the one of us who really did the work 😉

**Comment by philippwindischhofer** [comment]
Authorship also fixed here now.
